tests/test_subtyping_processing.py

```diff
@@ -95,3 +97,16 @@ def test_find_inconsistent_subtypes():
     inconsistent_subtypes = sorted_subtype_ints(pd.Series(subtypes_list))
     assert set(find_inconsistent_subtypes(inconsistent_subtypes)) == set(subtypes_list), \
         f'All subtypes should be inconsistent with each other in {subtypes_list}'
+
+def test_subtype_regex():
+    assert SubtypeCounts._check_subtype('x', 'x', '1.1.1.1') == '1.1.1.1'
```

Comment:
Thanks @DarianHole! :+1:  These are good checks showing that the regex you came up with is robust and can effectively catch invalid subtypes much better than the original solution. Would you be able to add one more check for internal periods, e.g. `1.1..1.1`? And maybe one test to show that any non-digit characters prefixed or suffixed to a valid subtype would be invalid (e.g. `a1.1.1` or `1.1.1a`)?

I think the code in this test could be refactored a bit to reduce duplication. 

You might be able to place the expected good values into a list and iterate your `assert`ions:

```python
for good_value in good_values:
    assert SubtypeCounts._check_subtype('x', 'x', good_value) == good_value
``` 

You could probably do the same thing with the values that would fail the check. That way any new valid or invalid cases could be added to a list rather than adding another line of code. 

Thanks!

Reply:
That is a great idea and looks a lot nicer then the way I had it previously. I need to remember to make these loops for the repetitive tasks.

Reply:
Yeah, the test looks really good and it's obvious what and how you're testing as well as how to extend the unit test in the future. :+1: